babel/messages/pofile.py

```diff
@@ -449,9 +451,13 @@ def _write_message(message, prefix=''):
             _write_comment(comment, prefix='.')
 
         if not no_location:
-            locs = u' '.join([u'%s:%d' % (filename.replace(os.sep, '/'), lineno)
-                              for filename, lineno in message.locations])
-            _write_comment(locs, prefix=':')
+            locs = []
+            for filename, lineno in message.locations:
+                if lineno:
+                    locs.append(u'%s:%d' % (filename.replace(os.sep, '/'), lineno))
```

Comment:
The `filename.replace(os.sep, '/')` duplication could be moved to a `filename = filename.replace(os.sep, '/')` statement.